### PR TITLE
Cache scikit-learn wheel in cross version tests

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Cache `scikit-learn` wheel to iterate faster in a PR that affects the scikit-learn flavor.

## How is this patch tested?

Existing tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/ml-package-versions.yml [hunk 1]
```diff
@@ -161,11 +164,10 @@ catboost:
     install_dev: |
       # The cross-version-tests workflow runs this command with the environment variable `CACHE_DIR`
       if [ ! -d "$CACHE_DIR" ] || [ -z $(find $CACHE_DIR -type f -name "catboost-*.whl") ]; then
-        head_sha=$(git ls-remote https://github.com/catboost/catboost.git HEAD | cut -f1)
         pip wheel --no-deps --wheel-dir $CACHE_DIR \
-          git+https://github.com/catboost/catboost.git@$head_sha#subdirectory=catboost/python-package
+          git+https://github.com/catboost/catboost.git#subdirectory=catboost/python-package
```
Comment (harupy) [inline mlflow/ml-package-versions.yml:168]: The latest version of `pip` automatically logs commit SHA when installing a package from a git repository: https://github.com/pypa/pip/pull/10149

File: mlflow/ml-package-versions.yml [hunk 2]
```diff
@@ -161,11 +164,10 @@ catboost:
...
       fi
-      pip install $(find $CACHE_DIR -type f -name "catboost-*.whl")
+      pip install $CACHE_DIR/catboost-*.whl
```
Comment (harupy) [inline mlflow/ml-package-versions.yml:170]: Simplified the command.

File: mlflow/ml-package-versions.yml [hunk 3]
```diff
@@ -287,14 +289,9 @@ spark:
         cd $temp_dir
         ./build/mvn -DskipTests --no-transfer-progress clean package
         cd python
-        python setup.py bdist_wheel
-
-        # Copy wheel in cache directory
-        wheel_path=$(find dist -type f -name "pyspark-*.whl")
-        mkdir -p $CACHE_DIR
-        cp $wheel_path $CACHE_DIR
+        python setup.py bdist_wheel --dist-dir $CACHE_DIR
```
Comment (harupy) [inline mlflow/ml-package-versions.yml:292]: Used `--dist-dir` to specify the output directory.

File: mlflow/ml-package-versions.yml (hunk 1 above)
Comment (dbczumar) [reply mlflow/ml-package-versions.yml:168]: Are we running the latest version of `pip`?

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

File: mlflow/ml-package-versions.yml (hunk 1 above)
Comment (harupy) [reply mlflow/ml-package-versions.yml:168]: @dbczumar Yes! https://github.com/mlflow/mlflow/blob/c80598208c63ae74d87f2011bafb651c9a843fb7/.github/workflows/cross-version-tests.yml#L126

**Pull request merged**